frontend/testing/cypress/src/integration/studio/datamodel.js

```diff
@@ -44,12 +44,18 @@ context('datamodel', () => {
     datamodel.getTypeField().invoke('val').should('eq', texts['schema_editor.integer']);
   });
 
-  it('Allows to upload an XSD file and displays the data model when the upload is complete', () => {
+  it('Allows to upload and then delete an XSD file', () => {
     cy.findAllByTestId(testids.fileSelectorInput)
       .first()
       .selectFile('src/fixtures/testdatamodel.xsd', { force: true });
     cy.findByRole('combobox', { name: texts['schema_editor.choose_model'] })
       .invoke('val')
-      .should('match', /\/testdatamodel.schema.json$/);
+      .should('match', /\/testdatamodel.schema.json$/)
+      .then((value) => {
+        cy.findByRole('combobox', { value }).should('exist');
+        cy.findByRole('button', { name: texts['schema_editor.delete_data_model'] }).click();
+        cy.findByRole('button', { name: texts['schema_editor.confirm_deletion'] }).click();
+        cy.findByRole('combobox', { value }).should('not.match', /\/testdatamodel.schema.json$/);
```

Comment:
Vi må sjekke at filen ikke lengre finnes i listen. Derfor foreslo jeg denne testen:
`cy.findByRole('option', { value }).should('not.exist')`
For å forsikre oss om at dette ikke gir et falskt positivt resultat, bør vi også ha en motsatt sjekk før vi utløser endringen:
`cy.findByRole('option', { value }).should('exist')`
Da blir det slik:
```suggestion
        cy.findByRole('option', { value }).should('exist');
        cy.findByRole('button', { name: texts['schema_editor.delete_data_model'] }).click();
        cy.findByRole('button', { name: texts['schema_editor.confirm_deletion'] }).click();
        cy.findByRole('option', { value }).should('not.exist');
```
Det fungerer i hvert fall lokalt hos meg. Feiler det hos deg?

`cy.findByRole('combobox', { value }).should('not.match', /\/testdatamodel.schema.json$/)` gir ikke så mye mening, fordi `value` _skal_ matche regexen her (det sjekker vi på linje 53). Så det er egentlig litt pussig at dette ikke feiler. Enten burde det feile fordi det matcher, eller fordi elementet ikke finnes.

Reply:
Jeg antar at `not.match` blir `true` hvis elementet ikke finnes og at det er derfor testen blir bestått. Uansett er ikke dette nok til å sjekke at elementet ikke finnes i listen, det sjekker bare om det er en annen verdi som er valgt.

Reply:
Ja, den feiler sånn :


<img width="493" alt="image" src="https://github.com/Altinn/altinn-studio/assets/90609090/905f0fe5-4472-4c33-b29b-de34d2326baa">

Reply:
Ok, la oss ta en nærmere titt på mandag!

Reply:
Feilen ble fikset med å bruke `findAllByRole` isteden for `findByRole` i denne linjen:
`cy.findAllByRole('option', { value }).should('exist');`
jeg vet ikke om det er greit å brukke `All` her ? siden at feilen sier at det finnes flere elementer med rollen "option".

Reply:
Jeg vil gjerne se hvordan det ser ut hos deg, for hos meg fungerer det helt fint med `findByRole`. Det burde ikke være flere av den. Vi må finne ut hvor den/de andre er - er det i samme liste, eller et annet sted på siden?

Reply:
Jeg sitter og feilsøker dette nå og tror jeg er på rett spor. Det ser ikke ut til at selektoren tar hensyn til det man har oppgitt i `value`. Jeg skal grave litt i det.

Reply:
Det kan se ut som at vi må løse dette med standard Cypress-funksjoner, siden Testing Library ikke fungerer slik vi ønsker her. Kan du prøve å erstatte `cy.findByRole('option', { value })` med ``cy.get(`option[value="${value}"]`)``?

Reply:
erstattet og den funket 👍